### PR TITLE
chore: merge alpha to master

### PR DESCRIPTION
**Pull request by KaiVandivier**
This will release version 12 🎉  [LIBS-704](https://dhis2.atlassian.net/browse/LIBS-704)

Before merge:
- [x] This PR should be merged first: https://github.com/dhis2/app-platform/pull/896
- [x] (Update): fix some sonarqube warnings: https://github.com/dhis2/app-platform/pull/898
- [x] Something weird is happening where new release commits on alpha [change yarn.lock in a breaking way](https://github.com/dhis2/app-platform/pull/896) and need to be [fixed by running `yarn` again](https://github.com/dhis2/app-platform/commit/5b9221256919184b72af82c4fec6f9f40749cb0c) -- it would be nice to make sure this doesn't happen on master
    - [x] Deps seem to be subtly affected in each release, but seemed to start creating a problem [after the introduction of React 18](https://github.com/dhis2/app-platform/pull/897/commits/313daae925b903b490e2665a2b05ad7015f985af#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2de)
    - [x] Update: it seems to be happening from the `npm version ___` command that's used in the `semantic-release` action
    - [x] Fixing React deps in the example apps at least avoids an app version conflict after a new version is published

After the merge:
- [ ] Release blog post: https://github.com/dhis2/developer-portal/pull/167

[LIBS-704]: https://dhis2.atlassian.net/browse/LIBS-704?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/_sidebar.md
Comment (Topener) [inline docs/_sidebar.md:1]: Just checking in here, but this file is no longer being used for the docs, these should be added in the developer portal instead. I have an upcoming PR that actually removes this file.

File: cli/src/commands/init.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ const handler = async ({ force, name, cwd, lib }) => {
         reporter.info('Installing @dhis2/cli-app-scripts...')
         await exec({
             cmd: 'yarn',
-            args: ['add', '--dev', '@dhis2/cli-app-scripts'],
+            args: ['add', '--dev', '@dhis2/cli-app-scripts@alpha'],
```
Comment (kabaros) [inline cli/src/commands/init.js:163]: this should be updated to remove alpha?

Comment (KaiVandivier) [reply cli/src/commands/init.js:163]: Indeed 🙂 https://github.com/dhis2/app-platform/pull/896 

(I think I'll just consider this an approval, and go ahead and merge that 😁 )

File: docs/_sidebar.md
Comment (KaiVandivier) [reply docs/_sidebar.md:1]: That should be handled here after this is merged: https://github.com/dhis2/developer-portal/pull/167/files#diff-9a6a0fdafefb0eecffb77784bc94f5ad984062167b8bcb7c4a899c90f6f567e8

**Review by amcgee — APPROVED** [review]
🎉🎉

**Pull request merged**